lib/vector-core/src/event/mod.rs

```diff
@@ -147,6 +147,69 @@ fn decode_value(input: proto::Value) -> Option<Value> {
     }
 }
 
+impl From<&tracing::Event<'_>> for Event {
+    fn from(event: &tracing::Event<'_>) -> Self {
+        let now = chrono::Utc::now();
+        let mut maker = MakeLogEvent(LogEvent::default());
+        event.record(&mut maker);
+
+        let mut log = maker.0;
+        log.insert("timestamp", now);
+
+        let meta = event.metadata();
+        log.insert(
+            "metadata.kind",
+            if meta.is_event() {
+                Value::Bytes("event".to_string().into())
+            } else if meta.is_span() {
+                Value::Bytes("span".to_string().into())
+            } else {
+                Value::Null
+            },
+        );
+        log.insert("metadata.level", meta.level().to_string());
+        log.insert(
+            "metadata.module_path",
+            meta.module_path()
+                .map(|mp| Value::Bytes(mp.to_string().into()))
+                .unwrap_or(Value::Null),
+        );
+        log.insert("metadata.target", meta.target().to_string());
+
+        log.into()
+    }
+}
+
+#[derive(Debug)]
```

Comment:
Can probably also derive `Default` here to avoid the explicit `LogEvent::default` above.

Reply:
Good idea.

Reply:
Were you leaving this for future work?